### ISSUE TITLE
zig is not yet a drop in replacement for gcc / clang

### ISSUE THREAD

**Issue by jimlloyd** [ziglang/zig#17273]
### Zig Version

0.11.0

### Steps to Reproduce and Observed Behavior

gcc and clang both have support of the options `-print-prog-name` but zig 0.10.0 does not. For example:

```
zig -print-prog-name=ar
```

### Expected Behavior

Something equivalent to what gcc and clang do with this option. gcc and clang examples:

```
$ gcc -print-prog-name=ar
ar
$ clang -print-prog-name=ar
/usr/bin/ar
```

Ideally the zig install would provide a symlink `zig-ar -> zig` and `zig` would use argv[0] to notice that the desired result is `zig ar ...`.

Note that the `-print-prog-name` option should work for at least `ar`, `ranlib`.

FYI I had hoped [this](https://gitlab.kitware.com/mrexodia/cmake-zig/-/tree/proper-support?ref_type=heads) would be sufficient to try zig as a drop-in replacement for a large body of C++ code built with `cmake`, but `boost` fails to configure without support for `-print-prog-name`.

**Comment by andrewrk** [comment]
can you please complete the bug template? you are missing "observed behavior" and "expected behavior" as well (what should the `zig -print-prog-name=ar` command print?)

**Comment by jimlloyd** [comment]
Hi Andrew. Thank you for your quick reply! I don't seem able to edit the form input in my response. I see you have removed the `bug` label and attached the `proposal` label, which I agree is a better way to frame this issue.

While researching this in more depth I discovered that `zig cc` already includes the full `clang` set of options, which makes fulfilling the goal much simpler. I believe it may be sufficient to simply provide symlinks to `zig` for `zig-cc`, `zig-ar`, `zig-ranlib`, and then make a simple enhancement to the `zig` CLI handling to recognize when `zig` was executed via the symlink by inspecting argv[0].

BTW `zig` looks awesome. I am looking for ways to start working with it. The path you have outlined in at least one talk to simply start with using `zig` as a drop-in replacement c/c++ compiler seems great. I have already switched to `bun` for my `node/javascript` work and I have ideas for things I would like to try using `bun`'s FFI. I'm also excited to see the support for WASM which I have some familiarity with.

**Comment by andrewrk** [comment]
Most of this proposal is a duplicate of #8716 and #8973. However, perhaps there is one more additional thing being proposed here, which is to make `zig cc -print-prog-name=ar` print something like `zig ar` rather than `/usr/bin/ar` or similar which is what it currently does.

**Comment by jimlloyd** [comment]
Yes, I agree that the existing issues #8716 and #8973 largely make this request redundant. I really like #8973 as it covers the cross compile use cases that I had glossed over here. 

Please note that `zig cc -print-prog-name=ar` should emit a single name. Emitting `zig ar` with the space will break various systems that expect a single name or path. This point was mentioned directly in [the first comment](https://github.com/ziglang/zig/issues/8716#issue-881220274) of #8716. However, that issue is closed so perhaps this point does need to be captured in #8973.

Please feel free to close this issue.
